src/Compilers/CSharp/Test/Syntax/Parsing/CollectionExpressionParsingTests.cs

```diff
@@ -16857,4 +16857,114 @@ public void InvokedCollectionExpressionVersusLocalFunctionAmbiguity14A()
         }
         EOF();
     }
+
+    [Fact, WorkItem("https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1934136")]
+    public void ByteArrayAmbiguityWithAttributes()
+    {
+        UsingTree("class C { public ReadOnlySpan<byte> B => [0, 1, 2, 3, 4, 5, 6, 7]; }");
```

Comment:
Do we still have a problem when a large collection-expr syntactically looks like a valid attribute list, e.g. `[a, b, c, d, e, f, ...]`

Do we still have a problem when the start of the collection-expr looks like an attribute list? e.g. `[x, 2, 3, 4, 5, 6, ...]`

Reply:
> Do we still have a problem when a large collection-expr syntactically looks like a valid attribute list, e.g. [a, b, c, d, e, f, ...]

No!  In that case, attribute parsing doesn't go 'off the rails'.  so we blitz through.  We do throw away rthe work.  Not ideal... but extremely hard to avoid.

> Do we still have a problem when the start of the collection-expr looks like an attribute list? e.g. [x, 2, 3, 4, 5, 6, ...]

Yes!  We would have the same issue then.  However, this PR wasn't an attempt to solve all the cases.  Just hte very realistic case of just numbers.  Basically: embedded binary blobs.  

We could def expand this further in the future though, esp. based on customer need.  THe idea then would be to make attribute parsing better about 'giving up' once things got bad enough.  But i think we can hold off on that for now.